src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIPhysicalHandle.cs

```diff
@@ -83,12 +83,19 @@ public override void ReturnPacket(SNIPacket packet)
 #if DEBUG
         private string GetStackParts()
         {
-            return string.Join(Environment.NewLine,
-                Environment.StackTrace
-                .Split(new string[] { Environment.NewLine }, StringSplitOptions.None)
-                .Skip(3) // trims off the common parts at the top of the stack so you can see what the actual caller was
-                .Take(7) // trims off most of the bottom of the stack because when running under xunit there's a lot of spam
-            );
+            // trims off the common parts at the top of the stack so you can see what the actual caller was
+            // trims off most of the bottom of the stack because when running under xunit there's a lot of spam
+            string[] parts = Environment.StackTrace.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);
+            List<string> take = new List<string>(7);
+            for (int index = 0; take.Count < 7 && index < parts.Length; index++)
+            {
+                if (index > 2)
```

Comment:
Why not start counting from index 3 immediately?

Reply:
How so? I see that 
```
for (int index = 3; take.Count < 7 && index < parts.Length; index++)
{
    take.Add(parts[index]);
}
```
would be unrolled like that
```
int index = 3;
start_loop:
if (take.Count < 7 && index < parts.Length)
{
    take.Add(parts[index]);
    index++;
    goto start_loop;
}

```

Reply:
Agreed, I was wrong. Changed.